crates/vm/levm/src/precompiles.rs

```diff
@@ -185,18 +197,26 @@ const FP2_ZERO_MAPPED_TO_G2: [u8; 256] = [
 pub const G1_POINT_AT_INFINITY: [u8; 128] = [0_u8; 128];
 pub const G2_POINT_AT_INFINITY: [u8; 256] = [0_u8; 256];
 
-pub fn is_precompile(address: &Address, fork: Fork) -> bool {
+pub fn is_precompile(address: &Address, fork: Fork, vm_type: VMType) -> bool {
     // Cancun specs is the only one that allows point evaluation precompile
     if *address == POINT_EVALUATION_ADDRESS && fork < Fork::Cancun {
         return false;
     }
-    // Prague or newers forks should only use this precompiles
+    // Prague or newers forks should only use these precompiles
     // https://eips.ethereum.org/EIPS/eip-2537
     if PRECOMPILES_POST_CANCUN.contains(address) && fork < Fork::Prague {
         return false;
     }
 
-    PRECOMPILES.contains(address) || PRECOMPILES_POST_CANCUN.contains(address)
+    // P256 verify Precompile only existed on L2 before Osaka
+    if address == &P256_VERIFICATION_ADDRESS && fork < Fork::Osaka && matches!(vm_type, VMType::L1)
```

Comment:
can you reorder this if? put first the fork, then match, then address, this puts the cheaper comparison first and short circuits if possible